docs/Debugging.md

```diff
@@ -0,0 +1,58 @@
+# Debugging Goblint
+
+## Building a Debuggable Goblint Executable
+
+To build a Goblint executable with debug information, run the following command within the `analyzer` directory.
+
+`make debug`
+
+## Debugging Goblint with VS Code
+
+To debug OCaml programs, you can use the command line interface of `ocamldebug` or make use of the Visual Studio Code
+integration provided by `hackwaly.ocamlearlybird`. In the following, we describe the steps necessary to set up this VS Code extension to
+debug Goblint.
+
+### Setting-up Earlybird
+
+Install the [`hackwaly.ocamlearlybird` extension](https://marketplace.visualstudio.com/items?itemName=hackwaly.ocamlearlybird) in your installation of Visual Studio Code.
+To be able to use this extension, you additionally need to install `ocamlearlybird` on the opam switch you use for Goblint. To do this, run the following command in the `analyzer` directory:
+
+`opam install earlybird`
+
+### Providing a Launch Configuration
+
+To let the `hackwaly.ocamlearlybird` extension know which executable it should debug, and which arguments it should pass, we have to provide a configuration file.
+The configuration file has to be named `launch.json` and must reside in the `./.vscode` directory. Here is an example `launch.json`:
+
+```JSON
+{
+    "version": "0.2.0",
+    "configurations": [
+      {
+        "name": "Goblint",
+        "type": "ocamlearlybird",
+        "request": "launch",
+        "program": "${workspaceFolder}/goblint.byte",
+        "arguments": [
+          "tests/regression/00-sanity/01-assert.c",
+          "--enable", "ana.int.interval",
+        ],
+        "stopOnEntry": false,
+      }
+    ]
+}
+```
+Note that the individual strings in `arguments` cannot contain spaces.
```

Comment:
I guess they can, but then such argument would be passed as if it was quoted in the shell to contain a literal space.